FoodBookApp/UI/Views/Login/LoginView.swift

```diff
@@ -64,15 +65,26 @@ struct LoginView: View {
                             locationService.requestLocationAuthorization()
                             schedule()
                         }
-                        
-                    } catch {
-                        print(error)
                     }
                 }
                 
             }
+            .disabled(networkService.isUnavailable || networkService.isLowConnection)
             .padding()
+            .colorMultiply(networkService.isUnavailable ? .gray : .white)
+            .alert(viewModel.errorMsg, isPresented: $viewModel.showAlert) {
+                Button("OK", role: .cancel) { }
+            }
             
+            if(networkService.isUnavailable) {
+                Text("No connection, please make sure you have internet access before attempting to log-in")
+                    .foregroundStyle(.red)
+            }
+            
+            if(networkService.isLowConnection) {
```

Comment:
I have a question, shouldn't this be else if? (I might be wrong tho)

Reply:
I think the two conditions will never both be true at the same time. One is if there is no network connection at all which is set with the path.status of the network monitor and the other is directly a boolean of the path which is `isConstrained`. From the docs and testing, this flag is only true when you set your celullar options to Low Data Mode.

Reply:
I can make the change so the second part of the message is shared among the two conditions, and leave as if's for now. I would still like to do better testing of these EC things to check edge cases and such so I would make the change as a bugfix.